src/documents/tests/test_consumer.py

```diff
@@ -1,14 +1,83 @@
 import re
 
-from django.test import TestCase
+from django.test import TestCase, override_settings
+from uuid import uuid4
+import os
+import shutil
 from unittest import mock
 from tempfile import TemporaryDirectory
+from django.conf import settings
 
 from ..consumer import Consumer
 from ..models import FileInfo, Tag
 
 
 class TestConsumer(TestCase):
+    SAMPLE_FILES = os.path.join(os.path.dirname(__file__), "samples")
+    deletion_list = []
+
+    def add_to_deletion_list(self, dirname):
+        self.deletion_list.append(dirname)
+
+    def setUp(self):
+        storage = "/tmp/paperless-tests-{}".format(str(uuid4())[:8])
+        os.makedirs(os.path.join(storage, "documents", "originals"))
+        os.makedirs(os.path.join(storage, "documents", "thumbnails"))
+        storage_override = override_settings(MEDIA_ROOT=storage)
+        storage_override.enable()
+        self.add_to_deletion_list(storage)
+
+        tmpdir = "/tmp/paperless-tests-{}".format(str(uuid4())[:8])
+        tmpdir_override = override_settings(CONVERT_TMPDIR=tmpdir)
+        tmpdir_override.enable()
+        self.add_to_deletion_list(tmpdir)
+
+    def tearDown(self):
+        for dirname in self.deletion_list:
+            shutil.rmtree(dirname, ignore_errors=True)
+
+    @override_settings(PAPERLESS_FILENAME_FORMAT="{correspondent}/{title}")
+    def test_file_consumption(self):
+        with TemporaryDirectory() as tmpdir:
+            myConsumer = Consumer(consume=tmpdir)
+
+            # Put sample document into consumption folder
+            shutil.copyfile(os.path.join(self.SAMPLE_FILES, "letter.pdf"),
+                            os.path.join(tmpdir, "letter.pdf"))
+
+            myConsumer.consume_new_files()
+
+            # Check if consumed file has been stored correctly
+            self.assertEqual(os.path.isfile(os.path.join(
+                settings.MEDIA_ROOT, "documents", "originals", "none",
+                "letter-0000001.pdf.gpg")), True)
+
+    @override_settings(PAPERLESS_FILENAME_FORMAT="{correspondent}/dummy")
+    def test_duplicate_file_consumption(self):
+        with TemporaryDirectory() as tmpdir:
+            myConsumer = Consumer(consume=tmpdir)
```

Comment:
Shouldn't we override `setting.SCRATCH_DIR` too or pass it to the consumer with `scratch=` here? Otherwise, we would create the default `/tmp/paperless` and might leave some files behind if we die.

Reply:
Yep, good idea. Unfortunately, this required a number of follow-up changes in other files (see commits) but in general I think it is good cleaning this up properly.